### PR TITLE
AP_Motors: add new PWM_TYPE_ANGLE

### PR DESCRIPTION
**Pull request by IamPete1**
This adds a new PWM type for angle output, this means motors respect MIN/MAX/TRIM and reversals. https://github.com/ArduPilot/ardupilot/pull/26300

### CODE REVIEW

File: libraries/AP_Motors/AP_Motors_Class.h
Comment (amilcarlucas) [inline libraries/AP_Motors/AP_Motors_Class.h:373]: Move }; to a new line

**Review by amilcarlucas — CHANGES_REQUESTED** [review]

**Comment by Williangalvani** [comment]
Not a big fan of the "angle" naming. but that is what it thinks it is doing under the hood, so...

It does work as expected after applying #26422, thanks!

File: libraries/AP_Motors/AP_Motors_Class.cpp
Comment (Williangalvani) [inline libraries/AP_Motors/AP_Motors_Class.cpp:189]: this feels odd. what is the expected range for `set_angle()`?

Comment (IamPete1) [reply libraries/AP_Motors/AP_Motors_Class.cpp:189]: You can set whatever you like. Here we just set the scale to match the range PWM input. We convert a 1000 to 2000 range to -500 to 500, so we set a angle of 500.

The whole range and angle thing is left over from before we did scaled values as floats, now we could just hard code everything to 1, but to maintain existing behavior we use random ranges, plane control surfaces use angle of 4500, plane throttle uses range of 100. Its all a bit odd.

Comment (Williangalvani) [reply libraries/AP_Motors/AP_Motors_Class.cpp:189]: oh I see it now. If I read the code correctly, `SRV_Channels::set_angle()` should be called `SRV_Channels::set_maximum_angle()`

Comment (Williangalvani) [reply libraries/AP_Motors/AP_Motors_Class.cpp:189]: which also explains why it is being used analogously to set_range()

**Review by Williangalvani — APPROVED** [review]
lgtm and worked as expected here, thanks!

But as I'm no maintainer for this code, I'll let a responsible adult do a review/merge.

File: libraries/AP_Motors/AP_Motors_Class.cpp
Comment (tridge) [inline libraries/AP_Motors/AP_Motors_Class.cpp:110]: this is subtracting two unsigned numbers, pwm could be 1000, offset will be 1500, so will wrap, then it gets cast to float. I'm not at all confident this will work, and may work differently in SITL to STM32

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Motors/AP_Motors_Class.cpp
Comment (IamPete1) [reply libraries/AP_Motors/AP_Motors_Class.cpp]: `_motor_pwm_scaled.offset` changed to float and `pwm` now cast to float.

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
@Williangalvani will test before merging

**Review by rmackay9 — APPROVED** [review]

**Comment by Williangalvani** [comment]
re-tested and working as expected. thanks @IamPete1 !

**Pull request merged**
